diff --git a/pintos/src/threads/thread.c b/pintos/src/threads/thread.c
index fadb363..b1ce255 100644
--- a/pintos/src/threads/thread.c
+++ b/pintos/src/threads/thread.c
@@ -504,7 +504,7 @@ next_thread_to_run (void)
     list_sort(&(ready_list), &compare_effective_priority, NULL);
     /*********/
 
-    return list_entry (list_pop_front (&ready_list), struct thread, elem);
+    return list_entry (list_pop_back (&ready_list), struct thread, elem);
 }
 
 /* Completes a thread switch by activating the new thread's page
